src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder salaryInfo = new StringBuilder();
+        int salary;
+        LocalDate start = LocalDate.parse(dateFrom, formatter); // converting String to Date
+        LocalDate end = LocalDate.parse(dateTo, formatter); // converting String to Date
+        salaryInfo.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        for (String name : names) { // take each name in turn from the array
+            salary = 0; // reset variable after every exit from the loop
+            for (String datum : data) {
+                /* split string by spaces and take first part as currentDate */
+                LocalDate currentDate = LocalDate.parse(datum.split(" ", 0)[0], formatter);
+                /* split string data[i] by spaces and take second part as name of employee */
+                String nameUser = datum.split(" ", 0)[1];
+                String workHours = datum.split(" ", 0)[2];
+                String incomePerHour = datum.split(" ", 0)[3];
+                if (currentDate.isAfter(start) && currentDate.isBefore(end)
+                        && nameUser.equals(name)) {
+                    salary += Integer.parseInt(workHours) * Integer.parseInt(incomePerHour);
+                } else if (currentDate.isEqual(start) && nameUser.equals(name)
+                        || currentDate.isEqual(end) && nameUser.equals(name)) {
+                    salary += Integer.parseInt(workHours) * Integer.parseInt(incomePerHour);
+                }
+            }
+            salaryInfo.append(System.lineSeparator())
+                    .append(name)
+                    .append(" - ")
+                    .append(salary);
+        }
+        return salaryInfo.toString();
```

Comment:
Remember to give informative names to variables and methods. In this context, 'salaryInfo' could be more descriptive, such as 'reportBuilder' or 'salaryReport', to better reflect its purpose.